lib/rules/no-array-prototype-extensions.js

```diff
@@ -152,12 +152,29 @@ function variableNameToWords(name) {
  * @param {Object} fixer The ESLint fixer object which will be used to apply fixes.
  * @returns {Object|[]}
  */
-function applyFix(callExpressionNode, fixer) {
+function applyFix(callExpressionNode, fixer, context) {
   const propertyName = callExpressionNode.callee.property.name;
 
   switch (propertyName) {
     case 'any':
       return fixer.replaceText(callExpressionNode.callee.property, 'some');
+    case 'filterBy': {
+      const callArgs = callExpressionNode.arguments;
+      const sourceCode = context.getSourceCode();
+
+      return [
+        fixer.replaceText(callExpressionNode.callee.property, 'filter'),
+        fixer.insertTextBefore(sourceCode.ast, "import { get } from '@ember/object';\n"),
+        fixer.replaceText(
+          callArgs[0],
+          `item => get(item, ${sourceCode.getText(callArgs[0])}) === ${sourceCode.getText(
+            callArgs[1]
+          )}`
+        ),
+        fixer.removeRange([callArgs[0].range[1], callArgs[1].range[0]]),
+        fixer.remove(callArgs[1]),
```

Comment:
Can you add a comment in front of each of these fixer lines explaining what each does?